src/axom/slic/docs/sphinx/sections/architecture.rst

```diff
@@ -0,0 +1,440 @@
+
+Slic provides a simple and easy to use logging interface for applications.
+
+.. _figs/slic_architecture:
+
+.. figure:: ../figures/architecture.png
+   :align: center
+   :scale: 75%
+   :alt: Slic Components Architecture
+
+   Basic Component Architecture of Slic.
+
+The basic component architecture of Slic, depicted in
+:numref:`figs/slic_architecture`, consists of three main components:
+
+#. A static logger API. This serves as the primary interface to the application.
+
+#. One or more Logger object instances. In addition to the root logger, which
+   is created when Slic is initialized, an application may create additional
+   loggers. However, at any given instance, there is only a single *active*
+   logger to which all messages are logged.
+
+   * A Logger consists of *four* log message levels: ``ERROR``,
+     ``WARNING``, ``INFO`` and ``DEBUG``.
+
+   * Each :ref:`logMessageLevel` can have one or more :ref:`logStream` instances,
+     which, specify the output destination, format and filtering of
+     messages.
+
+#. One or more :ref:`logStream` instances, bound to a particular logger, that
+   can be shared between log message levels.
+
+The application logs messages at an appropriate :ref:`logMessageLevel`, i.e.,
+``ERROR``, ``WARNING``, ``INFO``, or, ``DEBUG`` using the static API.
+Internally, the Logger routes the message to the corresponding
+:ref:`logStream` instances that are bound to the associated
+:ref:`logMessageLevel`.
+
+The following sections discuss some of these concepts in more detail.
+
+.. _logMessageLevel:
+
+Log Message Level
+^^^^^^^^^^^^^^^^^^
+
+The :ref:`logMessageLevel` indicates the severity of a message. Slic provides
+four levels of messages ranked from highest to lowest as follows:
+
++---------------------+--------------------------------------------------------+
+|   Message Level     |            Usage Description                           |
++=====================+========================================================+
+| ``ERROR``           |  Indicates that the application encountered a critical |
+|                     |  error or a faulty state. Includes also stacktrace.    |
++---------------------+--------------------------------------------------------+
+| ``WARNING``         |  Indicates that the application encountered an error,  |
+|                     |  but, the application should proceed.                  |
++---------------------+--------------------------------------------------------+
+| ``INFO``            |  General information reported by an application.       |
++---------------------+--------------------------------------------------------+
+| ``DEBUG``           |  Information useful to application developers.         |
++---------------------+--------------------------------------------------------+
+
+.. note::
+
+  ``ERROR`` messages by default will cause the application to abort. This
+  behavior may be toggled by calling ``slic::enableAbortOnError()`` and
```

Comment:
There is also a way to treat warnings as errors, right?

Reply:
There is a way to `abort` on warnings. Do we really need that? I am not sure anyone is actually using and I find it a bit questionable as a feature.

Reply:
Not sure...

I think it would be useful to be able to toggle the ability to dump the stacktrace in warnings though. It could help pinpoint the cause of warnings when the message doesn't give a hint about the caller.